tests/js/plugin/enhancedanchoring-test.coffee

```diff
@@ -634,3 +657,64 @@ describe 'Annotator.Plugin.EnhancedAnchoring', ->
             anchor.realize()
 
             assert anchor.fullyRealized
+
+    describe 'when scrolling to a virtual anchor', ->
+
+      it 'scrolls right next to the wanted page (to get it rendered)', ->
+        am = createAnchoringManagerAndLazyDocument()
+        ann = createTestAnnotationForPages "a1", [10]
+        anchor = am.createAnchor(ann, ann.target[0]).result
+        am.document.currentIndex = 5  # We start from page 5
+        am.document.setPageIndex = sinon.spy (index) ->
+          am.document.currentIndex = index
+          if index is 9
+            renderPage am.document, 9
+            renderPage am.document, 10
+
+        # Now we trigger the actual action
+        anchor.scrollIntoView().then ->
+          assert.calledWith am.document.setPageIndex, 9
```

Comment:
Okay, so what I don't understand here is why `setPageIndex` needs to be a spy that renders the page when all we are doing is asserting that the method itself is called with the correct value. Is the following not equivalent?

```coffee
it 'scrolls right next to the wanted page (to get it rendered)', ->
   am = createAnchoringManagerAndLazyDocument()
   ann = createTestAnnotationForPages "a1", [10]
   anchor = am.createAnchor(ann, ann.target[0]).result

   am.document.currentIndex = 5  # We start from page 5
   am.document.setPageIndex = sandbox.stub()

   # Now we trigger the actual action
   anchor.scrollIntoView().then ->
     assert.calledWith am.document.setPageIndex, 9
```